### ISSUE TITLE
Not writing (or potentially reading) bit rasters correctly

### ISSUE THREAD

**Issue by lossyrob** [locationtech-labs/geopyspark#191]
I ingested a bit raster (the Chattanooga data) successfully, but when I tried to read a value out of the layer it gave an error like this:

```
error: cannot reshape array of size 8192 into shape (256,256) - Traceback (most recent call last):
  File "/home/hadoop/.local/lib/python3.4/site-packages/geonotebook/vis/geotrellis/server.py", line 50, in tile
    return fn(z, x, y)\n  File "/home/hadoop/.local/lib/python3.4/site-packages/geonotebook/vis/geotrellis/server.py", line 89, in tile
    arr = ser.loads(tile._1())[0][\'data\']\n  File "/home/hadoop/.local/lib/python3.4/site-packages/geopyspark/avroserializer.py", line 92, in loads
    return [self.decoding_method(schema_dict)]\n  File "/home/hadoop/.local/lib/python3.4/site-packages/geopyspark/avroregistry.py", line 41, in tile_decoder
    arrs = [cls._tile_decoder(band) for band in bands]\n  File "/home/hadoop/.local/lib/python3.4/site-packages/geopyspark/avroregistry.py", line 41, in <listcomp>
    arrs = [cls._tile_decoder(band) for band in bands]\n  File "/home/hadoop/.local/lib/python3.4/site-packages/geopyspark/avroregistry.py", line 20, in _tile_decoder
    arr = np.array(cells).reshape(schema_dict[\'rows\'], schema_dict[\'cols\'])\nValueError: cannot reshape array of size 8192 into shape (256,256)
```

If I do the ingest like this:

```python
def ingest_layer(layer_name, path):
    rdd = get(geopysc, SPATIAL, path, maxTileSize=512, numPartitions=20)
    metadata = rdd.collect_metadata(tile_size=512)
    metadata['cellType'] = "int8"
    laid_out = rdd.tile_to_layout(metadata)
    reprojected = laid_out.reproject("EPSG:3857", scheme=ZOOM)
    for layer_rdd in reprojected.pyramid(reprojected.zoom_level, 0):
        write(catalog_uri, layer_name, layer_rdd)    
```

it works.

**Comment by jbouffard** [comment]
I actually discovered this bug a few days ago, but failed to report it. Sorry about that. PR #166 fixes this.

**Comment by jbouffard** [comment]
After looking more into this issue, it appears that there isn't a way to represent 1 bit values in a numpy array. Instead, it would have to be stored as an array of `bool`s or as 1's and 0's. The problem with this is that these values would then be stored as a byte each. See this [list](https://docs.scipy.org/doc/numpy/user/basics.types.html#id7) of all the types that numpy supports.

According to [this](https://stackoverflow.com/questions/5602155/numpy-boolean-array-with-1-bit-entries) StackOverflow question, there are other means of representing this array in Python; however, this would make the API for bit arrays different from all of the other types, so using another representation probably isn't the way to go.

We might not be able to support the decoding of `BitArrayTile`s in GeoPySpark. I don't think this will be a major issue unless someone tries to reserialize the array.

What is your opinion on this @echeipesh ?

**Comment by jbouffard** [comment]
Resolved in #245
